### PR TITLE
feat: `ARGO_TEMPLATE` env var without `inputs.parameters`

### PR DESCRIPTION
**Pull request by tooptoop4**
Fixes https://github.com/argoproj/argo-workflows/issues/8790

**Motivation**

reduce etcd growth

**Modifications**

has env variable to maintain existing functionality for some users

### CODE REVIEW

File: docs/environment-variables.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ This document outlines environment variables that can be used to customize behav
...
 | `ARGO_AGENT_MEMORY_LIMIT`                | `resource.Quantity` | `256m`                                                                                      | Memory resource limit for the agent.                                                                                                                                                                                                                                     |
 | `ARGO_POD_STATUS_CAPTURE_FINALIZER`      | `bool`              | `false`                                                                                     | The finalizer blocks the deletion of pods until the controller captures their status.
+| `ARGO_TEMPLATE_WITH_INPUTS`      | `bool`              | `true`                                                                                     | Whether to keep large inputs section inside the environment variable of pods.
```
Comment (jswxstw) [inline docs/environment-variables.md:26]: I think `ARGO_TEMPLATE_WITH_INPUTS_PARAMETERS` is more semantically appropriate and it is better to write it clearly in the introduction to keep `inputs.parameters` in the `ARGO_TEMPLATE` environment variable.

**Comment by juliev0** [comment]
Did you test it with both environment variable set "false" and unset?

**Comment by juliev0** [comment]
As a follow up to your various changes, would you be able to add to the docs the various ways that somebody can reduce their overall Pod size, including through [this](https://github.com/argoproj/argo-workflows/pull/12325) method?

**Comment by tooptoop4** [comment]
> Did you test it with both environment variable set "false" and unset?

yes

> As a follow up to your various changes, would you be able to add to the docs the various ways that somebody can reduce their overall Pod size, including through [this](https://github.com/argoproj/argo-workflows/pull/12325) method?

i can raise a follow up docs pr

**Review by juliev0 — APPROVED** [review]

File: docs/environment-variables.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This document outlines environment variables that can be used to customize behav
...
 | `ARGO_AGENT_MEMORY_LIMIT`                | `resource.Quantity` | `256m`                                                                                      | Memory resource limit for the agent.                                                                                                                                                                                                                                     |
 | `ARGO_POD_STATUS_CAPTURE_FINALIZER`      | `bool`              | `false`                                                                                     | The finalizer blocks the deletion of pods until the controller captures their status.
+| `ARGO_TEMPLATE_WITH_INPUTS_PARAMETERS`      | `bool`              | `true`                                                                                     | Whether to keep inputs.parameters inside the ARGO_TEMPLATE environment variable of pods.
```
Comment (agilgur5) [inline docs/environment-variables.md:26]:
```suggestion
| `ARGO_TEMPLATE_WITH_INPUTS_PARAMETERS`      | `bool`              | `true`                                                                                     | Whether to keep `inputs.parameters` inside the `ARGO_TEMPLATE` environment variable of pods
```

These should have backticks, and then the `.spelling` entries are not needed.
That it pops up as a spelling issue can indicate a different issue, like lack of backticks.

Comment (tooptoop4) [reply docs/environment-variables.md:26]: https://github.com/argoproj/argo-workflows/pull/13761 addresses this

**Pull request merged**